src/helpers/ApiServiceHelper.js

```diff
@@ -29,8 +28,124 @@ function ApiServiceHelper (url = '', clientId = '', clientSecret = '', scope = '
     };
   };
 
+  this.handleErrorsByStatusCode = (error, record, serviceName, cb) => {
+    const functionName = 'handleErrorsByStatusCode';
+    const errorType = serviceName.replace(/\s+/g, '-').toLowerCase() + '-error';
+    let errorMessage = `an error was received from the ${serviceName} for HoldRequestId: ${record.id}`;
+
+    if (error.response) {
+      const statusCode = error.response.status;
+      const statusText = error.response.statusText.toLowerCase() || '';
+      let errorTypeByService = '';
+      let errorMessageByService = '';
+
+      if (serviceName === 'Patron Service') {
+        errorTypeByService = 'hold-request-record-missing-patron-data';
+        errorMessageByService = `unable to retrieve Patron data from Patron Service for hold request record (${record.id})`;
+      }
+
+      if (serviceName === 'Item Service') {
+        errorTypeByService = 'hold-request-record-missing-item-data';
+        errorMessageByService = `unable to retrieve Item data from Item Service for hold request record (${record.id})`;
+      }
+
+      errorMessage += `; ${statusText}`;
+
+      switch (statusCode) {
+        case 400:
+        case 404:
+          // POST FAILURE TO STREAM
+          logger.error(
```

Comment:
Should you maybe move this out this particular `case` and log for every status code? Right now, I don't think we'd have a log for, say, 500 error messages.